### PR TITLE
Add the permissions resource from 2947605

### PR DESCRIPTION
**Pull request by dawehner**
Pulls in the permission patch from https://www.drupal.org/project/drupal/issues/2947605

### CODE REVIEW

**Comment by dawehner** [comment]
That is a great idea! I'll have a look at it later today.

**Comment by tedbow** [comment]
@dawehner I pushed up some changes to this branch. 
see new commits

File: src/Plugin/rest/resource/PermissionsCollectionResource.php [hunk 1]
```diff
@@ -18,7 +18,7 @@
  *   id = "permissions_collection",
  *   label = @Translation("Available permissions"),
  *   uri_paths = {
- *     "canonical" = "/permissions"
+ *     "canonical" = "/admin-api/permissions"
```
Comment (dawehner) [inline src/Plugin/rest/resource/PermissionsCollectionResource.php]: +1

Comment (dawehner) [inline src/Plugin/rest/resource/PermissionsCollectionResource.php]: +1

File: src/Plugin/rest/resource/PermissionsCollectionResource.php [hunk 2]
```diff
@@ -3,6 +3,7 @@
 namespace Drupal\admin_ui_support\Plugin\rest\resource;
 
 use Drupal\Component\Render\PlainTextOutput;
+use Drupal\Core\Extension\ModuleHandlerInterface;
```
Comment (dawehner) [inline src/Plugin/rest/resource/PermissionsCollectionResource.php]: Note: The module handler is actually the wrong service. I fixed it in some additional commit.

**Pull request merged**